storybookjs/storybook
# Vue3: Use slot value directly if it's a string in source decorator

## What I did



Support plain text for "slot" controls. 

Currently, only functions `() => "Default Text Content"`, VNode `h("p", "Default Text Content")`, and stringified objects are supported. 

and even if we do use those supported ways, the control table will display something like


```
// Function, renders as `Default Text Content`
default: {
}

// VNode, renders as `<p>Default Text Content</p>`
footer: {
  __v_isVNode: true,
  ...,
  type: "p",
  ...
  children: "Default Text Content",
  ...,
}

// Object, not even renders, but shows `{ a: 1, b: 2 }` in the "Show Code"
header: {
  a: 1,
  b: 2,
}
```

But I want users to be able to change the slot content, if it's only a label, so if the slot content is a string, we just render that instead.



ALSO, without this change, if you use a String slot content, on the "Show Code" block, you'll only see an empty template tag.

```html
<template #default></template>
```


## How to test


1. Run a sandbox for template. I ran `yarn task --task sandbox --start-from auto --template vue3-vite/default-ts`
2. Open Storybook in your browser
3. Access Stories >  renderers > vue3_vue3-vite-default-ts > Reactive Slots  > Docs
4. See aforementioned behavior on the slots controls.
5. Edit any of the slots by clicking Raw button and Updating it with a double quoted string.
6. Unfocus on the input
7. The canvas should reflect the current string value, including the "Show Code" block. 


## Checklist



	I don't know where to write tests, the slot function isn't tested anywhere.
	I don't think it is documented anywhere, but if I missed it, please let me know.
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
